command/agent/keyring_endpoint.go

```diff
@@ -0,0 +1,140 @@
+package agent
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// KeyringRequest is used route operator/raft API requests to the implementing
+// functions.
+func (s *HTTPServer) KeyringRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
```

Comment:
Note that the RFC was missing (until just now) the URL paths for this HTTP API, so I'm totally open to changing those.